pkg/storegateway/bucket_chunk_reader.go

```diff
@@ -242,6 +243,132 @@ type loadIdx struct {
 	chunk       int
 }
 
+type rangeLoadIdx struct {
+	offset uint32
+	estLen int64
+	// seriesEntry and rangeEntry are indices for where to put the result,
+	// they are not actual refs of series or chunks
```

Comment:
[nit] A little clarification.

```suggestion
	// they are not actual refs of series or chunks in the block.
```